### ISSUE TITLE
Blindness on login not working correctly

### ISSUE THREAD

**Issue by Dragonber1** [PaperMC/Paper#3362]
### What behaviour is expected:
Player logs in and blindness effect is applied to his view instantly

### What behaviour is observed:
Player logs in and blindness effect is applied but takes between 7 to 15 seconds to be active on the players view.

### Steps/models to reproduce:
Using any authentication plugin that has an option to enable a blindness effect on login, like AuthMe or loginSecurity. Using any plugin that can enable a potion effect on login like Infinite Potions.

### Plugin list:
Authme, LuckPerms, ProtocolLib

### Paper build number: 
280

### Anything else:
I believe this could be related to async login, or login priorities

**Comment by MiniDigger** [comment]
it just depend on when the client recieves and processes this packet.
I have similar "issues" after a teleport for example.
I don't think there is anything paper can do here.

**Comment by aikar** [comment]
If they are sending it during PlayerLoginEvent instead of PlayerJoinEvent, then i could see it, but that's abug in the plugin, they should be using Join event.

**Comment by Dragonber1** [comment]
As an alternative I setup a plugin to give blindness to the player via console command. I tested a range of 1 to 10 tick delay to the command after the player joins. The tests where not very accurate, but they showed that there where inconsistencies when giving the effect during the first 5 ticks. The effect would be given, the player could see its particles, but he would not be blind until 5 or 6 ticks later. Blindness always worked when giving the effect 6 or more ticks after joining.

At this point I believe that this is just like what MiniDigger said and its a client side issue, however, I still find it strange that the player can see blindness particles being applied correctly on join but not the blindness effect. Another concern of mine is that all the plugins that I've tried which are able to apply this effect on login have this same behavior.
